coffee-jpa/src/main/java/hu/icellmobilsoft/coffee/jpa/sql/batch/BatchService.java

```diff
@@ -19,873 +19,1062 @@
-//        return dbTimeZone;
-//    }
-//
-//    /**
-//     * Returns the {@link ZoneOffset} by {@link ZoneId}.
-//     * 
-//     * @param zoneId
-//     *            the {@link ZoneId}.
-//     * @return {@link ZoneOffset} by {@link ZoneId}.
-//     */
-//    protected ZoneOffset getZoneOffset(ZoneId zoneId) {
-//        return OffsetDateTime.now(zoneId).getOffset();
-//    }
-//
-//    private <E> void executeBatch(Map<String, Status> result, PreparedStatement ps, List<String> tmpProcessedEntities) throws SQLException {
-//        int[] executeBatchResult = ps.executeBatch();
-//        addBatchResult(result, tmpProcessedEntities, executeBatchResult);
-//    }
-//
-//    private void addBatchResult(Map<String, Status> result, List<String> entityIds, int[] batchResult) {
-//        if (entityIds == null || entityIds.isEmpty()) {
-//            return;
-//        } else if (entityIds.size() < batchResult.length) {
-//            throw new IllegalArgumentException("Each batchResult must have an associated entityId!");
-//        }
-//
-//        for (int i = 0; i < entityIds.size(); i++) {
-//            Status status;
-//            if (i < batchResult.length) {
-//                int resultCode = batchResult[i];
-//                status = getStatus(resultCode);
-//            } else {
-//                // kevesebb batchResult jött vissza, mint entityId (pl Oracle hajlamos erre, ha a batch utolsó eleme failel el, és log error reject
-//                // limittel fut), nem tudjuk mi lett a hiányzó rekorddal, UNKNOWN státuszt jelölünk
-//                status = Status.UNKNOWN;
-//            }
-//            result.put(entityIds.get(i), status);
-//        }
-//    }
-//
-//    private Status getStatus(int resultCode) {
-//        if (resultCode > 0) {
-//            return Status.SUCCESS.setRowsAffected(resultCode);
-//        } else {
-//            switch (resultCode) {
-//            case 0:
-//                return Status.SUCCESS_NO_UPDATE;
-//            case Statement.SUCCESS_NO_INFO:
-//                return Status.SUCCESS_NO_INFO;
-//            case Statement.EXECUTE_FAILED:
-//                return Status.EXECUTE_FAILED;
-//            default:
-//                return Status.UNKNOWN;
-//            }
-//        }
-//    }
+    /** Constant <code>BATCH_SIZE=100</code> */
+    public static final int BATCH_SIZE = 100;
+
+    @Inject
+    @ThisLogger
+    private AppLogger log;
+
+    @SuppressWarnings("cdi-ambiguous-dependency")
+    @Inject
+    private EntityManager entityManager;
+
+    private String sqlPostfix;
+
+    private TimeZone dbTimeZone;
+
+    /**
+     * input validalas.
+     * 
+     * @param entities
+     *            - validalni kivant collection
+     * @throws BaseException
+     *             exception
+     */
+    protected void validateInput(Collection<?> entities) throws BaseException {
+        if (entities == null) {
+            log.warn("entities is null skipped to save!");
```

Comment:
a save helyett processe-et írnék vagy valami jó általános dolgot, mert lehet itt törlés is akár.